client/src/components/ApplicantInbox.js

```diff
@@ -0,0 +1,40 @@
+import React, { useState, useEffect } from "react";
+import { getMessagesByApplicantId } from "../api/messages";
+import { Segment, Message } from "semantic-ui-react";
+
+const ApplicantInbox = () => {
+  const user = JSON.parse(localStorage.getItem("user"));
+  const [messages, setMessages] = useState([]);
+
+  useEffect(() => {
+    const fetchMessages = async () => {
+      try {
+        const response = await getMessagesByApplicantId(user.id);
+        setMessages(response || []);
+      } catch (error) {
+        setMessages([]);
+      }
+    };
+    if (user.role === "applicant" && !messages.length) {
+      fetchMessages();
```

Comment:
If you change this to `user && user.role === "applicant" && !messages.length` then this ensures a slightly more graceful failure of just seeing an empty inbox if no user is logged in. Otherwise you will get an error that you can't access property `role` of `null` (cf. truthiness and falsiness, in JS `null` is a falsy value which means it evaluates to false in conditionals). It will only evaluate later parts of the conditional if `user` has a valid value).

Reply:
Done. Please review. Thanks

Reply:
Thanks.